### ISSUE TITLE
[Frontend] Add GUI with input fields for range + CTAs for scanning network

### ISSUE THREAD

**Issue by vinitshahdeo** [vinitshahdeo/PortScanner#12]
Currently this is a CLI based application, convert this into a **GUI application** to make the UX better!

- _**Must have:**_ 
     - **A window with menus** (IP Scanning/Port Scanning)
     - **Input box to enter host**
     - **Input fields to enter low & high range of ports**
     - **CTA(Button) to start scanning**
     - **List of Ports/IPs** (once the scanning is done)

- **_Good to have:_**
     - A live timer to show the total scanning time
     - Some kind of indication (animation) in UI while scanning is happening in the background
     - Resume/Pause/Restart scanning options

- **_Extended feature_**
     - UI option to convert a host into its IP address
     - Display history: Last five scans in the following tabular form

|  Date & Time  |     Host     |  IP Address  | Total Scanning time |
|:--------:|:------------:|---|---|
| 9:00 PM June 8, 2020 | localhost | 127.0.0.1  | 24 seconds |

> Recommended to use [PyQt5](https://pypi.org/project/PyQt5/) | Learn more [here](https://build-system.fman.io/pyqt5-tutorial).

**PS: Still open for discussions - feel free to suggest GUI frameworks(preferably from Python).**

**Comment by vinitshahdeo** [comment]
@Kashish121 Changed the directory structure and updated the steps in `README.md`. Please have a look and verify that nothing is breaking and one can easily run the project following the steps mentioned in `README.md`. 

@ishika1727 Please rebase your `master` branch and raise the PRs again. Sorry for your inconvenience. 😞

**Comment by Kashish121** [comment]
On it.

**Comment by ishika1727** [comment]
On it.

**Comment by Kashish121** [comment]
No issues. PortScanner can be run easily following the steps mentioned in `README.md`. 
Error due to director structure #19 also resolved.

**Comment by vinitshahdeo** [comment]
**TODO<sup>as discussed in meeting</sup>**

@ishika1727 - Explore [materialize](https://materializecss.com/)
@Kashish121  - Explore [flask](https://www.fullstackpython.com/flask.html).

Post your thoughts here for the wireframe.

**Comment by vinitshahdeo** [comment]
From now onwards, we'll track our progress [here](https://github.com/users/vinitshahdeo/projects/9). 

#### @Kashish121 , @ishika1727 - [Check out](https://github.com/users/vinitshahdeo/projects/9) the Project board.

**Comment by ishika1727** [comment]
@vinitshahdeo  Please review this UI

![image](https://user-images.githubusercontent.com/42978414/84580883-ff9a0500-adf8-11ea-92a1-4ffd7c0d3e45.png)

**Comment by vinitshahdeo** [comment]
@ishika1727 Few suggestions from my end:

- The CTA (`Scan Now`) should be at the end.
- The input text fields shouldn't cover the whole width.
- The sidebar should be fixed a bit - it looks like an admin portal now. Remove email, just have logo!
- Add helper texts for all the inputs.

> PS: Tagging @NainikaB here, she's the best person for UI/UX. Nainika, please post your thoughts here. For your context, it's a SPA for scanning available ports in the given network<sup>to be scaled to go-to networking tool with multiple features</sup>, learn more [here](https://vinitshahdeo.github.io/PortScanner/).

**Comment by ishika1727** [comment]
@vinitshahdeo I have made the required changes. Please review it.

![image](https://user-images.githubusercontent.com/42978414/84603305-fc5f5180-aeaa-11ea-806c-e2d5a1b7cb4c.png)

**Comment by nainikaB** [comment]
@vinitshahdeo thank you for tagging me! :) Yes, I'd love to! And you have already made some key observations regarding the functionality of it. 

@ishika1727 Okay so a couple of things, I think the page lacks design. I do understand that this is just a utility tool but, maybe give the user an experience and interphase they'll always remember.

So instead of having it route to contact and about, there's actually ample space to accommodate all of your content in the same page, of course with a smart use of drop downs in the mobile UI. So maybe consider that as well :) 

For the current design: 
 - Logo placement issues with regards to size and position. 
- Consider making it a box so you won't feel like you have to use the whole desktop width
- Darken the colour for the headings host to scan, low and high. It currently looks like it's disabled.
- Consider using bold for the font and changing the font-family 
- What's the current font size? it look its 16-18px. Maybe increase it?
- Is that email id for users to contact you incase of an issue?
- I think there should be social media links on this page itself, if you are considering having social media presence for this. 

And lastly, I would suggest using a wireframing tool or software like figma when you guys are going back and forth with designs since there are bound to be many versions before you actually settle with one and it puts way too much work on the person building it. I don't know if you guys have already done that but this is just a suggestion it would make working as a team and collaborating a lot more smoother.

**Comment by vinitshahdeo** [comment]
@nainikaB Wow, this is an intensive review! ❤️ 
Thanks for your valuable thoughts. We'll definitely look into these. ✌️ 

> PS: We're not using Figma or any wireframing tools due to time limitation and less UI work. Ideally we should have done it. Please keep an eye here, we'll keep you posted.

**Comment by vinitshahdeo** [comment]
> @vinitshahdeo I have made the required changes. Please review it.
> 
> ![image](https://user-images.githubusercontent.com/42978414/84603305-fc5f5180-aeaa-11ea-806c-e2d5a1b7cb4c.png)


@ishika1727, **Center align the GitHub logo in sidebar and let's change the color of icon to the theme color.** Rest we can discuss over the call.

**Comment by nainikaB** [comment]
@vinitshahdeo ayeee please don't thank me, happy to be of any kind of help with your project. :) Although, I think I might have increased the work load, sorry! If you guys are on a strict time constraint ignore the complete rearranging of elements that I suggested and instead, maybe improvise with the current design itself along with the comments I have attached above and anything else that I might have missed. And yes, I'll definitely keep an eye here and help you guys with anything else you might need :)  Yes that's true, Figma does take time definitely not advisable when you have a deadline approaching.

**Comment by nainikaB** [comment]
> > @vinitshahdeo I have made the required changes. Please review it.
> > ![image](https://user-images.githubusercontent.com/42978414/84603305-fc5f5180-aeaa-11ea-806c-e2d5a1b7cb4c.png)
> 
> @ishika1727, **Center align the GitHub logo in sidebar and let's change the color of icon to the theme color.** Rest we can discuss over the call.
Please change the font for the title/heading.

**Comment by vinitshahdeo** [comment]
@nainikaB 😊  Will ensure that the UI bit includes all of your suggestions. Thanks for being kind, as always!

**Keep guiding!** ✌️

**Comment by vinitshahdeo** [comment]
**[ Frontend / TODO ]**

@ishika1727 Here's sharing a few utilities which you can use to finalize the fonts. Use a combo of **three** font-families - **heading**, **sub-heading** and **paragraph**.

- [typespiration.com](https://typespiration.com/design/) (Recommended)
- [fontjoy.com](https://fontjoy.com/)
- [fontpair.co](https://fontpair.co/)

Check [colorhunt.co](https://colorhunt.co/) to get inspiration for color palletes.

@Kashish121 Join her and finalize the font and theme color for the website.

**Comment by vinitshahdeo** [comment]
**[ Deliverables by Sunday EOD ]**

- **Updated UI with all the changes** (including font-family and final theme color) - @ishika1727
     - Create a `feature/frontend` branch from `master` and push your changes. (Follow [CONTRIBUTING.md](https://vinitshahdeo.github.io/PortScanner/CONTRIBUTING.html))
     - Create a PR for the same

- **Frontend to be rendered using Flask** (We can look for adding actions on button-click later) - @Kashish121
    - Create a `feature/backend-flask` branch from `master` and push your changes (Follow [CONTRIBUTING.md](https://vinitshahdeo.github.io/PortScanner/CONTRIBUTING.html))
    - Create a PR for the same

> PS: Lemme know if there're any blockers around this.

**Comment by Kashish121** [comment]
@vinitshahdeo, Working on it already. 👍 Will share the updates.

**Comment by ishika1727** [comment]
> **[ Deliverables by Sunday EOD ]**
> 
> * **Updated UI with all the changes** (including font-family and final theme color) - @ishika1727
>   
>   * Create a `feature/frontend` branch from `master` and push your changes. (Follow [CONTRIBUTING.md](https://vinitshahdeo.github.io/PortScanner/CONTRIBUTING.html))
>   * Create a PR for the same
> * **Frontend to be rendered using Flask** (We can look for adding actions on button-click later) - @Kashish121
>   
>   * Create a `feature/backend-flask` branch from `master` and push your changes (Follow [CONTRIBUTING.md](https://vinitshahdeo.github.io/PortScanner/CONTRIBUTING.html))
>   * Create a PR for the same
> 
> > PS: Lemme know if there're any blockers around this.

Doing the required changes. Will create a PR soon

**Comment by vinitshahdeo** [comment]
@ishika1727 & @Kashish121 - Any updates? Let's sync up and finish the pending tasks.

**Comment by Kashish121** [comment]
Hey @vinitshahdeo , got the required changes in the UI today. Will push the the code for connectivity tomorrow.

**Comment by ishika1727** [comment]
@vinitshahdeo already created a PR yesterday with the changes you suggested.

**Comment by vinitshahdeo** [comment]
@ishika1727 Added few comments on #39

**Comment by vinitshahdeo** [comment]
@Kashish121 / @ishika1727 - Please try running [this](https://github.com/vinitshahdeo/PortScanner/tree/feature/frontend) locally and confirm me.

> Please follow the [steps](https://github.com/vinitshahdeo/PortScanner/tree/feature/frontend) mentioned on `README.md`

**Comment by Kashish121** [comment]
Works for me @vinitshahdeo . @ishika1727 can you confirm once?

**Comment by ishika1727** [comment]
works for me too

**Comment by Sohaib03** [comment]
Is this open ? I would like to make an GUI interface using PyQt5.

**Comment by vinitshahdeo** [comment]
Hey @Sohaib03, thanks for showing your interest. Please go ahead.

PS: The website is done but let's have Desktop GUI too.

**Comment by Sohaib03** [comment]
I will be using PyQt5 with Python 3. Is that okay ?

**Comment by vinitshahdeo** [comment]
@Sohaib03 That's fine. Just ensure that you branch off from correct branch (`feature/python3`) and follow the recommended directory structure.

**Comment by Sohaib03** [comment]
How is this for the UI ? 
![ss](https://user-images.githubusercontent.com/16899515/87636517-af4cf480-c762-11ea-89de-a86336161646.png)

**Comment by vinitshahdeo** [comment]
@Sohaib03 This looks great. 🤩  👏  
Super excited to run this locally once you **raise the PR**. 

cc: @Kashish121 / @ishika1727

**Comment by Sohaib03** [comment]
Thanks. I am working on it.

**Comment by Sohaib03** [comment]
I want to make a pull request. Do you want to review the code ?

**Comment by vinitshahdeo** [comment]
@Sohaib03 Yes, please raise the PR. 
Not only me, @Kashish121 and @ishika1727  will also review the code.
